examples/shapley_value_with_customized_featurizers.ipynb

```diff
@@ -0,0 +1,1176 @@
+     "output_type": "stream",
+     "text": [
+      "Prediction with keep_features set 0: tensor([2.2535])\n",
+      "Prediction with keep_features set 1: tensor([2.2319])\n",
+      "Prediction with keep_features set 2: tensor([2.1880])\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/oscarwu/code/chemprop_developing/chemprop/data/dataloader.py:52: UserWarning: Dropping last batch of size 1 to avoid issues with batch normalization (dataset size = 1, batch_size = 64)\n",
+      "  warnings.warn(\n",
+      "/home/oscarwu/mambaforge/envs/chemprop_v2/lib/python3.12/site-packages/lightning/fabric/plugins/environments/slurm.py:204: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python /home/oscarwu/mambaforge/envs/chemprop_v2/lib/python ...\n",
+      "/home/oscarwu/mambaforge/envs/chemprop_v2/lib/python3.12/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:424: The 'predict_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=63` in the `DataLoader` to improve performance.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# example prediction with different keep_atom_features and keep_bond_features\n",
+    "\n",
+    "# keep all atom and bond features\n",
+    "keep_atom_features_0 = [True]*8\n",
+    "keep_bond_features_0 = [True]*4\n",
+    "\n",
+    "# keep some atom and bond features\n",
+    "keep_atom_features_1 = [True, True, False, True, True, False, True, True]\n",
+    "keep_bond_features_1 = [True, True, False, True]\n",
+    "\n",
+    "# remove all atom and bond features\n",
+    "keep_atom_features_2 = [False]*8\n",
+    "keep_bond_features_2 = [False]*4\n",
+    "\n",
+    "pred_0 = get_predictions(keep_atom_features_0, keep_bond_features_0, test_mol)\n",
+    "pred_1 = get_predictions(keep_atom_features_1, keep_bond_features_1, test_mol)\n",
+    "pred_2 = get_predictions(keep_atom_features_2, keep_bond_features_2, test_mol)\n",
+    "\n",
+    "print(f\"Prediction with keep_features set 0: {pred_0}\") # expected 2.2535\n",
+    "print(f\"Prediction with keep_features set 1: {pred_1}\") # expected 2.2319\n",
+    "print(f\"Prediction with keep_features set 2: {pred_2}\") # expected 2.1880"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 197,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# An example wrapper class for use as the model input in SHAP explainer\n",
+    "# The wrapper needs to be initialized first with the molecule to be explained, and then can be called with a boolean list representing the features to keep\n",
+    "# The wrapper is needed because SHAP explainer requires a callable model with a single input argument, adapt X as needed\n",
+    "class MoleculeModelWrapper:\n",
```

Comment:
Stylistic suggestion, this could be function instead of a class. 
```
def get_predictions_with_features(mol, X):
    preds = []
    for keep_features in X:
        try:
            # unpacking X, indices corresponds to feature orders from default chemprop featurizer, adapt as needed
            keep_atom_features = keep_features[:8] # 8 atom features
            keep_bond_features = keep_features[8:] # 4 bond features
        except:
            print(f"Invalid input: {keep_features}")
            raise
        pred = get_predictions(keep_atom_features, keep_bond_features, mol)
        preds.append([pred.item()])
    return np.array(preds)
model = lambda X: get_predictions_with_features(test_mol, X)
```